pages/view/_iscnId/index.vue

```diff
@@ -82,57 +82,19 @@
       'lg:px-0',
     ]"
   >
-    <div
-      :class="[
-        'flex',
-        'justify-end',
-        'items-center',
-        'w-full',
-        'pt-[24px]',
-        'gap-[8px]',
-        'lg:pt-0',
-      ]"
-    >
-      <div v-if="isIscnOwner" class="flex justify-center items-center gap-[4px] mr-[8px]">
-        <Button preset="plain" text="Edit ISCN" @click="handleEdit">
-          <template #prepend>
-            <IconEdit />
-          </template>
-        </Button>
-        <div class="h-[20px] w-[2px] bg-medium-gray" />
-      </div>
-      <Button preset="tertiary" :text="$t('NFTPortal.button.download.iscn')" @click="handleClickDownload">
-        <template #prepend>
-          <IconDownload />
-        </template>
-      </Button>
-      <Button
-        v-if="isShowMintButton && isNFTBook"
-        preset="secondary"
-        class="w-full lg:w-auto"
-        :text="$t('NFTPortal.button.mint.book')"
-        @click="clickMintNFTBook"
-      />
-      <Button
-        v-if="isShowMintButton && !isNFTBook"
-        preset="secondary"
-        class="w-full lg:w-auto"
-        :to="localeLocation({ name: 'nft-iscn-iscnId', params: { iscnId: iscnId }, query: mintQueries })"
-        :text="$t('NFTPortal.button.mint')"
-      />
-      <div v-if="!!classId" class="flex justify-center items-center p-[4px] rounded-[12px] border-like-cyan-light border-[2px]">
-        <Button
-          preset="tertiary"
-          :text="$t('NFTPortal.button.check.nft')"
-          :href="likerlandNftUrl"
-          target="_blank"
-        >
-          <template #append>
-            <IconArrowRight />
-          </template>
-        </Button>
-      </div>
-    </div>
+    <IscnEditBar
+      :is-iscn-owner="isIscnOwner"
+      :iscn-id="iscnId"
+      :is-show-mint-button="isShowMintButton"
+      :is-nft-book="isNFTBook"
+      :mint-queries="mintQueries"
```

Comment:
What does this do? Can we just get route query in component instead of passing prop?

Reply:
[🐛 Fix missing data](https://github.com/likecoin/app-like-co/pull/438/commits/07e3a680b226ced8adb15fbe7c9a918873e4eb24)